nexus_constructor/main_window.py

```diff
@@ -198,3 +198,41 @@ def show_add_component_window(self, component: Component = None):
         )
         self.add_component_window.ui.setupUi(self.add_component_window)
         self.add_component_window.show()
+
+
+class QDialogCustom(QDialog):
+    """
+    Custom QDialog class that enables the possibility to properly produce
+    a message box in the component editor to the users,
+    asking if they are sure to quit editing component when exiting.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._is_accepting_component = True
+
+    def enable_msg_box(self):
+        self._is_accepting_component = True
```

Comment:
This is also done in the constructor. Do we need them both?

Reply:
Yes, you are correct. I was planning to remove it but forgot. :) As it is only relevant when closing the window, setting to true again makes no sense. I will remove it.